### PR TITLE
Apply NaN filtering while rendering series

### PR DESCRIPTION
**Pull request by rallen090**
**What this PR does / why we need it**:

Before, this was being done in a preceding filtering function which mutated the underlying series array.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/query/api/v1/handler/prometheus/native/common.go [hunk 1]
```diff
@@ -355,16 +308,41 @@ func RenderResultsJSON(
 			// would be at the result node but that would make it inefficient since
 			// we would need to create another block just for the sake of restricting
 			// the bounds.
-			if dp.Timestamp.Before(opts.Start) {
+			if dp.Timestamp.Before(opts.Start) || dp.Timestamp.After(opts.End) {
```
Comment (rallen090) [inline src/query/api/v1/handler/prometheus/native/common.go:311]: One inconsistency w/ the prev implementation is that the filter-fn would not count data after `End` towards a series that would be included, but then in the normal rendering iteration it wouldn't skip these. Feels more correct to apply this in that path.

File: src/query/api/v1/handler/prometheus/native/common.go [hunk 2]
```diff
@@ -328,25 +294,14 @@ func RenderResultsJSON(
...
-		for _, t := range s.Tags.Tags {
-			jw.BeginObjectField(string(t.Name))
-			jw.WriteString(string(t.Value))
-		}
-		jw.EndObject()
-
-		jw.BeginObjectField("values")
-		jw.BeginArray()
 
+		currentDatapoint := 0
```
Comment (arnikola) [inline src/query/api/v1/handler/prometheus/native/common.go:298]: nit; can be a bool since value is unused after incrementing once

**Review by arnikola — APPROVED** [review]
LGTM, potentially worth just removing the keep nans option entirely, but that can be in future

File: src/query/api/v1/handler/prometheus/native/common.go (hunk 2 above)
Comment (rallen090) [reply src/query/api/v1/handler/prometheus/native/common.go]: Done

**Pull request merged**